### PR TITLE
Make @ClientExceptionMapper work with Kotlin

### LINKED ISSUE #27309: Kotlin: Rest Client Reactive @ClientExceptionMapper not usable
### Describe the bug

When you try to define a [@ClientExceptionMapper](https://quarkus.io/guides/rest-client-reactive#using-clientexceptionmapper) with Kotlin like this:

```kotlin
@RegisterRestClient(baseUri = "https://swapi.dev/api")
interface SwapiRestClient {
    @GET
    @Path("/people/{id}")
    suspend fun getPersonById(@RestPath id: Int, @RestQuery format: SwapiFormat?): Person

    companion object {
        @JvmStatic
        @ClientExceptionMapper
        fun toException(response: Response): RuntimeException? {
            return when (response.status) {
                400 -> NotFoundException("not found")
                else -> null
            }
        }
    }
}
```

the application will fail on startup with the exception:

```
Caused by: java.lang.IllegalStateException: io.quarkus.rest.client.reactive.ClientExceptionMapper is only supported on static methods of REST Client interfaces that take 'javax.ws.rs.core.Response' as a single parameter and return 'java.lang.RuntimeException'.
	at io.quarkus.rest.client.reactive.deployment.ClientExceptionMapperHandler.generateResponseExceptionMapper(ClientExceptionMapperHandler.java:78)
```

However, the `@JvmStatic` annotation should create a normal static interface method in the bytecode since Kotlin 1.3 (see https://kotlinlang.org/docs/java-to-kotlin-interop.html#static-methods).

I can confirm the rest client interface method is usable from Java with a normal static access: 

```java
public class TestStaticAccess {

	public void testStatic() {
		RuntimeException ex = SwapiRestClient.toException(Response.accepted().build());
		if (ex != null) {
			throw ex;
		}
	}
}
```

### Expected behavior

It should be possible to use a rest client with `@ClientExceptionMapper` with Kotlin as shown above. If possible, it would be great to even check if the method exists on the Companion object so we could avoid using `@JvmStatic` and make the usage more Kotlin idiomatic. 

### Actual behavior

App does not start and shows exception that the annotated method would not be valid.

```
Caused by: java.lang.IllegalStateException: io.quarkus.rest.client.reactive.ClientExceptionMapper is only supported on static methods of REST Client interfaces that take 'javax.ws.rs.core.Response' as a single parameter and return 'java.lang.RuntimeException'.
	at io.quarkus.rest.client.reactive.deployment.ClientExceptionMapperHandler.generateResponseExceptionMapper(ClientExceptionMapperHandler.java:78)
```

### How to Reproduce?

Reproducer:
[2022-08-15-quarkus.zip](https://github.com/quarkusio/quarkus/files/9350320/2022-08-15-quarkus.zip)
 
1. Download and unzip
2. start with `./gradlew quarkusDev`
3. observe error

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

JDK 17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

Quarkus 2.11.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

gradle

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #27309

### CODE REVIEW

**Comment by geoand** [comment]
@brunobat @ebullient `io.quarkus.it.micrometer.prometheus.ClientRequestTest` seems to be flaky (I've seen it fail in other PRs as well).

**Comment by brunobat** [comment]
Will take a look @geoand

**Comment by geoand** [comment]
👌

File: extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/ClientExceptionMapperHandler.java [hunk 1]
```diff
@@ -54,8 +54,12 @@ Result generateResponseExceptionMapper(AnnotationInstance instance) {
...
         boolean isValid = false;
         if (instance.target().kind() == AnnotationTarget.Kind.METHOD) {
-            if ((instance.target().asMethod().flags() & Modifier.STATIC) != 0) {
-                targetMethod = instance.target().asMethod();
+            MethodInfo methodInfo = instance.target().asMethod();
+            if (ignoreAnnotation(methodInfo)) {
+                return null;
+            }
+            if ((methodInfo.flags() & Modifier.STATIC) != 0) {
+                targetMethod = methodInfo;
```
Comment (gastaldi) [inline extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/ClientExceptionMapperHandler.java:62]:
```suggestion
            targetMethod = instance.target().asMethod();
            if (ignoreAnnotation(targetMethod)) {
                return null;
            }
            if ((targetMethod.flags() & Modifier.STATIC) != 0) {
```

**Review by gastaldi — APPROVED** [review]

**Pull request merged**